.travis.yml

```diff
@@ -1,47 +1,40 @@
 language: python
+python: 2.7
 sudo: false
 cache:
   pip: true
   directories:
-    - $HOME/buildout-cache
+  - $HOME/.pylint.d
+  - eggs
 
 global:
   - secure: "nTXMNU2aYbCtvKD74PndF8Xm2h6IvsxPjr6vj45AxjXmaB7Wf5oGi/4b6ObV\nexBtUQkFnc+M0ThGyUznOCPi3YxcbTzsyhQYhHUtmtw/6QRyYgo4E0GLDItU\n37Ff6wpxl3NMMJTvpi6SOVUvIJUl3+cs+4bkMkS48RSFGg2rGmo="  # Coveralls.io token
+env:
+  matrix:
+  - PLONE_VERSION=4.3
+  - PLONE_VERSION=5.0
+  - PLONE_VERSION=5.1
 matrix:
   fast_finish: true
-  include:
-      # pyXX-plone_sunburst,pyXX-plone_barceloneta
-      - python: "2.7"
-        env: TOXENV=py27-plone-4.3
-      - python: "2.7"
-        env: TOXENV=py27-plone-5.0
-      - python: "2.7"
-        env: TOXENV=py27-plone-5.1
-      - python: "2.7"
-        env: TOXENV=isort,flake8,docs,
 
 before_install:
-    - mkdir -p $HOME/buildout-cache/{downloads,eggs,extends}
```

Comment:
this is what I call a "a test setup from old times"; caching has been around Travis since 2016.

Reply:
sure, tell me how to do it better?

if you use buildout and not the standard tools like pip which is known for travis you have to tell them which directory to cache:

https://github.com/plone/plone.api/blob/master/.travis.yml#L3-L6

So the maked line is necessary to be idempotent and deterministic.

Reply:
it isn't; if you check what I did you'll learn how to fix it. we have been using it that way for years and we already had that discussion in bobtemplates.plone as @jensens mentioned.